eclipse/src/saros/ui/command_handlers/SessionAddSelectedContactsHandler.java

```diff
@@ -1,23 +1,49 @@
 package saros.ui.command_handlers;
 
 import java.util.List;
-import org.eclipse.core.commands.AbstractHandler;
-import org.eclipse.core.commands.ExecutionEvent;
-import org.eclipse.core.commands.ExecutionException;
+import javax.inject.Named;
+import org.eclipse.e4.core.di.annotations.CanExecute;
+import org.eclipse.e4.core.di.annotations.Execute;
+import org.eclipse.e4.core.di.annotations.Optional;
+import org.eclipse.e4.ui.services.IServiceConstants;
+import saros.SarosPluginContext;
+import saros.communication.connection.ConnectionHandler;
 import saros.net.xmpp.JID;
+import saros.net.xmpp.contact.XMPPContactsService;
+import saros.repackaged.picocontainer.annotations.Inject;
 import saros.session.ISarosSession;
+import saros.session.ISarosSessionManager;
 import saros.ui.util.CollaborationUtils;
 import saros.ui.util.selection.retriever.SelectionRetrieverFactory;
 
 /** Handles the addition of selected {@link JID}s to the running {@link ISarosSession}. */
-public class SessionAddSelectedContactsHandler extends AbstractHandler {
+public class SessionAddSelectedContactsHandler {
 
-  @Override
-  public Object execute(ExecutionEvent event) throws ExecutionException {
+  @Inject private ConnectionHandler connectionHandler;
+  @Inject private ISarosSessionManager sessionManager;
+  @Inject private XMPPContactsService contactsService;
+
+  public SessionAddSelectedContactsHandler() {
+    SarosPluginContext.initComponent(this);
+  }
+
+  @Execute
+  public Object execute() {
 
     List<JID> jids = SelectionRetrieverFactory.getSelectionRetriever(JID.class).getSelection();
 
     CollaborationUtils.addContactsToSession(jids);
     return null;
   }
+
+  @CanExecute
+  public boolean canExecute(@Named(IServiceConstants.ACTIVE_SELECTION) @Optional JID contact) {
```

Comment:
I do not think that this is the correct implementation. IIRC only the host can invite other contacts.

Reply:
Nvm. The original code archive this by hiding the command instead of disabling the handler.

Reply:
I think was only ensured by the visibleWhen expression in the pop menu command and not in the handler.

https://github.com/saros-project/saros/blob/master/eclipse/plugin.xml#L664
And
https://github.com/saros-project/saros/blob/master/eclipse/plugin.xml#L1042